### PR TITLE
Add an Operator for Forest Inference Libary (FIL) Triton backend.

### PR DESCRIPTION
**Pull request by oliverholworthy**
Add an Operator for Forest Inference Libary (FIL) Triton backend.

- [x] figure out output signature automatically (e.g. classification, + with proba)
- [x] allow params to be passed in and take precedence over defaults

### CODE REVIEW

File: merlin/systems/dag/ops/fil.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import pathlib
+
+import numpy as np
+from google.protobuf import text_format  # noqa
+
+import merlin.systems.triton.model_config_pb2 as model_config  # noqa
```
Comment (karlhigley) [inline merlin/systems/dag/ops/fil.py:6]: In parallel with your PR, we just switched over to importing this from the tritonclient in #109 to avoid some issues with building/registering classes from Protobuf twice:
```suggestion
import tritonclient.grpc.model_config_pb2 as model_config # noqa
```

File: merlin/systems/dag/ops/fil.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    def __init__(self, model):
+        """Instantiate a FILPredict inference operator.
+
+        Parameters
+        ----------
+        model : xgboost.Booster
+            An XGBoost model.
+        """
+        self.model = model
+        self.input_schema = Schema([ColumnSchema("input__0", dtype=np.int32)])
```
Comment (karlhigley) [inline merlin/systems/dag/ops/fil.py:27]: Since we have methods to compute the schemas, might make sense to return these schemas from those methods (even though they're functionally constants in this case.)

File: merlin/systems/dag/ops/fil.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        self.output_schema = Schema([ColumnSchema("output__0", dtype=np.int32)])
```
Comment (karlhigley) [inline merlin/systems/dag/ops/fil.py:28]: Wondering if this should be a float type like `np.float32` but not completely sure what FIL expects.

File: tests/unit/systems/test_fil.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+import merlin.systems.dag.ops.fil as fil_op
+from merlin.dag import ColumnSelector, Graph
+from merlin.schema import Schema
+
+# this needs to be before any modules that import protobuf
+os.environ["PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION"] = "python"
+
+from google.protobuf import text_format  # noqa
+
+model_config = pytest.importorskip("nvtabular.inference.triton.model_config_pb2")
```
Comment (karlhigley) [inline tests/unit/systems/test_fil.py:32]:
```suggestion
model_config = pytest.importorskip("tritonclient.grpc.model_config_pb2")
```

File: .github/workflows/cpu-ci.yml [hunk 5]
```diff
@@ -33,6 +33,7 @@ jobs:
         python -m pip install --upgrade pip setuptools==59.4.0 wheel 
         python -m pip install -r requirements-dev.txt
         python -m pip install tensorflow
+        python -m pip install xgboost
```
Comment (karlhigley) [inline .github/workflows/cpu-ci.yml:36]: Now that we're supporting more than one modeling library, we should make `tensorflow` and `xgboost` optional dependencies. For this PR, probably good enough to make `xgboost` optional and we do the same for TF in a separate PR.

**Review by karlhigley — COMMENTED** [review]
Looks good! Left a few minor comments

File: .github/workflows/cpu-ci.yml (hunk 5 above)
Comment (oliverholworthy) [reply .github/workflows/cpu-ci.yml:36]: can you clarify what you mean by optional in the context of this line. The systems package currently only has two dependencies, merlin-core and nvtabular. Do you mean adding something like we have with models where we can pip install merlin-models[xgboost] ? 

the FIL op as currently written doesn't strictly depend on xgboost (import-wise), only in the sense that some calling it will need to load a model - maybe that could be left up to them to install?

File: merlin/systems/dag/ops/fil.py (hunk 3 above)
Comment (oliverholworthy) [reply merlin/systems/dag/ops/fil.py]: yes, FIL does expect float32

File: merlin/systems/dag/ops/fil.py (hunk 2 above)
Comment (oliverholworthy) [reply merlin/systems/dag/ops/fil.py]: that is a bit cleaner, updated

File: .github/workflows/cpu-ci.yml (hunk 5 above)
Comment (karlhigley) [reply .github/workflows/cpu-ci.yml:36]: Yeah, I think we might want to set up `pip install merlin-systems[xgboost]`, but it's out of scope for this PR. This line just reminded me that it's a thing to do.
